SM-1012: Removing SM Beta (Phase 2)

coltonhurst opened the pull request:
## Type of change

<!-- (mark with an `X`) -->

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
<!--Describe what the purpose of this PR is. For example: what bug you're fixing or what new feature you're adding-->

Now that phase 1 of SM beta is complete, in phase 2 we remove all references to SecretsManagerBeta in the server project, but leave the references in the db for phase 3. We need to do this to ensure compatibility between the current and previous versions of server with the database.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

Thomas-Avery reviewed (changes requested):
Looks good, just need to add EF migrations.

coltonhurst commented on src/Core/AdminConsole/Entities/Organization.cs:
```diff
@@ -78,7 +78,6 @@ public class Organization : ITableObject<Guid>, ISubscriber, IStorable, IStorabl
     public int? SmServiceAccounts { get; set; }
     public int? MaxAutoscaleSmSeats { get; set; }
     public int? MaxAutoscaleSmServiceAccounts { get; set; }
-    public bool SecretsManagerBeta { get; set; }
```
Good call, thank you! Fixed in [d6a8a0f](https://github.com/bitwarden/server/pull/3663/commits/d6a8a0facbd715ea4eff3a19af6079b44389fcd1)

Thomas-Avery reviewed (dismissed):
Looks good.

rkac-bw reviewed (approved):
looks good to me

vincentsalucci reviewed (approved):
Reviewed the AC team owned files. Quick question: how **good** did making this PR feel? 😄

coltonhurst commented:
@vincentsalucci @cturnbull-bitwarden @rkac-bw @Thomas-Avery I'll need a review one more time if you don't mind!

This has been approved by QA, but I re-ran the EF migrations so that the date would be updated. This should be the only change since the last review.

Thank you!

vincentsalucci reviewed (approved):
Approved AC file changes 👍

rkac-bw reviewed (approved):
Looks good to me
